packages/editor/src/components/collab-sidebar/comment-indicator-toolbar.js

```diff
@@ -0,0 +1,185 @@
+
+				// If we have more than 1 page, there are more comments
+				if ( totalPages && totalPages > 1 ) {
+					moreCommentsExist = true;
+				}
+
+				if ( comments ) {
+					// Get all comments in this thread.
+					// Main comment has id === blockCommentIdValue
+					// Replies have parent === blockCommentIdValue
+					const threadComments = comments.filter(
+						( comment ) =>
+							comment.status !== 'trash' &&
+							( comment.id === blockCommentIdValue ||
+								comment.parent === blockCommentIdValue )
+					);
+
+					// Sort by date to show participants in chronological order.
+					threadComments.sort(
+						( a, b ) => new Date( a.date ) - new Date( b.date )
+					);
+
+					// Find the main thread comment (first comment).
+					const mainComment = threadComments.find(
+						( comment ) => comment.id === blockCommentIdValue
+					);
+
+					// If no main comment is found, the thread doesn't exist.
+					if ( ! mainComment ) {
+						return {
+							threadParticipants: [],
+							hasUnresolved: false,
+							hasMoreComments: false,
+						};
+					}
+
+					// Thread is resolved if the main comment is approved.
+					isResolved = mainComment.status === 'approved';
+
+					threadComments.forEach( ( comment ) => {
+						// Track thread participants (original commenter + repliers)
+						if (
+							comment.author_name &&
+							comment.author_avatar_urls
+						) {
+							const authorKey = `${ comment.author }-${ comment.author_name }`;
+							if ( ! participantsMap.has( authorKey ) ) {
+								participantsMap.set( authorKey, {
+									name: comment.author_name,
+									avatar:
+										comment.author_avatar_urls?.[ '48' ] ||
+										comment.author_avatar_urls?.[ '96' ],
+									isOriginalCommenter:
+										comment.id === blockCommentIdValue,
+									date: comment.date,
+								} );
+							}
+						}
+					} );
+				}
+			}
+
+			// Convert to array and maintain chronological order.
+			const participants = Array.from( participantsMap.values() );
+
+			return {
+				threadParticipants: participants,
+				hasUnresolved: ! isResolved,
+				hasMoreComments: moreCommentsExist,
+			};
+		},
+		[]
+	);
+
+	if ( ! threadParticipants.length ) {
+		return null;
+	}
+
+	// Show up to 3 avatars, with overflow indicator.
+	const maxAvatars = 3;
+	const visibleParticipants = threadParticipants.slice( 0, maxAvatars );
+	const overflowCount = Math.max( 0, threadParticipants.length - maxAvatars );
+
+	// If we hit the comment limit, show "100+" instead of exact overflow count.
+	const overflowText =
+		hasMoreComments && overflowCount > 0 ? '100+' : `+${ overflowCount }`;
+
+	const overflowTitle =
+		hasMoreComments && overflowCount > 0
+			? '100+ participants'
+			: `+${ overflowCount } more participants`;
+
+	return (
+		<CommentIconToolbarSlotFill.Fill>
+			<ToolbarButton
+				className={ `comment-avatar-indicator ${
+					hasUnresolved ? 'has-unresolved' : 'all-resolved'
+				}` }
```

Comment:
```suggestion
				className={ clsx( 'comment-avatar-indicator', {
					'has-unresolved': hasUnresolved,
				} ) }
```

The `all-resolved` classname seems not to be used anywhere.

Reply:
I thought it might be better to already have classes that indicate the state, but yes its better to only keep the necessary classes